### DISCUSSION TITLE
Order of middlewares in Nuxt3 layers

### DISCUSSION THREAD

**Discussion by maestroosram** [nuxt/nuxt#25231] [Questions]
In my project I have a Nuxt layer B extending a Nuxt project A.

I noticed that `/middleware` files are called in alphabetical order as should be but first all the middlewares of B and then all the middlewares of A.

If for example I have in A the following middlewares:
`00.first.global.ts`
`01.second.global.ts`

And in B these ones:
`10.third.global.ts`
`11.fourth.global.ts`

The final ordering of calls will be:
`10.third.global.ts`
`11.fourth.global.ts`
`00.first.global.ts`
`01.second.global.ts`

Is this intentional? And if so, why? Imho a layer has higher specificity then the extended project, so It should always run its middlewares after, to eventually overwrite something from extended project too.

Is there a way to force the rearrangement of the middlewares?

**Comment by chrishuman0923** [comment]
Noticing the same behavior for server middleware. Agreed that layer middleware should have higher priority over consuming app. Hopefully someone from the Nuxt team can answer this.
